workers/generic-worker/multiuser.go

```diff
@@ -34,18 +34,16 @@ var (
 )
 
 func secure(configFile string) {
-	if !config.RunTasksAsCurrentUser {
-		secureError := fileutil.SecureFiles(configFile)
-		exitOnError(CANT_SECURE_CONFIG, secureError, "Not able to secure config file %q", configFile)
-	}
+	secureError := fileutil.SecureFiles(configFile)
+	exitOnError(CANT_SECURE_CONFIG, secureError, "Not able to secure config file %q", configFile)
 }
 
 func rebootBetweenTasks() bool {
 	return !config.HeadlessTasks
 }
 
 func PostRebootSetup(taskUserCredentials *gwruntime.OSUser) {
-	pd, err := process.NewPlatformData(config.RunTasksAsCurrentUser, config.HeadlessTasks, taskUserCredentials)
+	pd, err := process.NewPlatformData(config.HeadlessTasks, taskUserCredentials)
```

Comment:
I've forgotten how this `pd` is used, but this looks like it might introduce a change in behaviour. I wonder if pd should be part of the task struct, rather than a global now.